seal_fhe/src/key_generator.rs

```diff
@@ -354,6 +362,31 @@ impl SecretKey {
     }
 }
 
+impl PartialEq for SecretKey {
+    fn eq(&self, other: &Self) -> bool {
+        self.as_bytes() == other.as_bytes()
```

Comment:
Thanks for taking this on! I guess I overlooked this while making things serializable.

Move the `as_bytes()` implementation out from `impl SecretKey` to under `impl ToBytes for SecretKey`. This makes it compatible with `WithContext`.

`WithContext` is our wrapper to make artifacts compatible with Serde Serialize/Deserialize traits. It manages first deserializing the SealContext so it can be passed to `from_bytes` when deserializing the data payload.